geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/fs/storage/parquet/ParquetFileSystemStorage.scala

```diff
@@ -81,6 +83,32 @@ object ParquetFileSystemStorage {
       observer.write(f)
     }
     override def flush(): Unit = observer.flush()
-    override def close(): Unit = CloseQuietly(Seq(writer, observer)).foreach(e => throw e)
+    override def close(): Unit = {
+      println(s"Closing file ${file}")
+      CloseQuietly(Seq(writer, observer)).foreach(e => throw e)
+      validateParquetFile(file)
+    }
+  }
+
+  def validateParquetFile(file: Path): Unit = {
+    if (!fileValidationEnabled.get.toBoolean) {
+      return
+    }
+
+    val reader = ParquetReader.builder(new GroupReadSupport(), file).build()
+
+    try {
+      // Read Parquet file content
+      var record = reader.read()
+      while (record != null) {
+        // Process the record
+        record = reader.read()
+      }
+      println(s"${file} is a valid Parquet file")
+    } catch {
+      case _: Exception => throw new RuntimeException(s"Unable to validate ${file}: File may be corrupted")
```

Comment:
```suggestion
      case e: Exception => throw new RuntimeException(s"Unable to validate ${file}: File may be corrupted", e)
```
this will show the underlying error, which is often useful